jbpm-wb-integration/jbpm-wb-integration-backend/src/test/java/org/jbpm/workbench/wi/backend/server/workitem/ServiceTaskServiceImplTest.java

```diff
@@ -0,0 +1,134 @@
+        serviceTaskServiceImpl.setRepoService(repoService);
+        serviceTaskServiceImpl.setM2Repository(m2Repository);
+
+        File file = new File(ServiceTaskServiceImplTest.class.getResource("/workitem.test-1.0.jar").getFile());
+        when(m2Repository.getArtifactFileFromRepository(new GAV(GAV))).thenReturn(file);
+    }
+
+    private void loadServices(String name, List<RepoData> currentServices) {
+        service = new RepoData();
+        service.setName(name);
+        service.setModule("module");
+        service.getInstalledOn().add("myspace/myproject");
+
+        currentServices.add(service);
+        storage.synchronizeServices(currentServices);
+    }
+
+    @Test
+    public void testAddServiceTasksByCreated() {
+        List<RepoData> currentServices = new ArrayList<>();
+        loadServices("Test1", currentServices);
+        assertEquals(SERVICETASKSSIZE, serviceTaskServiceImpl.getServiceTasks().size());
+        Map<String, List<String>> resultMap = serviceTaskServiceImpl.addServiceTasks(GAV);
+        assertEquals(SERVICETASKSSIZE + 1, serviceTaskServiceImpl.getServiceTasks().size());
+        assertEquals(WORKITEMNAME, resultMap.get(RepoService.CREATED).get(0));
+        assertTrue(resultMap.get(RepoService.SKIPPED).isEmpty());
+        Optional optional = serviceTaskServiceImpl.getServiceTasks().stream().filter(serviceTaskSummary -> serviceTaskSummary.getName().equals(WORKITEMNAME)).findFirst();
+        assertTrue(optional.isPresent());
+    }
+
+    @Test
+    public void testAddServiceTasksBySkipped() {
+        List<RepoData> currentServices = new ArrayList<>();
+        loadServices("Test", currentServices);
+        assertEquals(SERVICETASKSSIZE, serviceTaskServiceImpl.getServiceTasks().size());
+        Map<String, List<String>> resultMap = serviceTaskServiceImpl.addServiceTasks(GAV);
+        assertEquals(SERVICETASKSSIZE, serviceTaskServiceImpl.getServiceTasks().size());
+        assertEquals(WORKITEMNAME, resultMap.get(RepoService.SKIPPED).get(0));
+        assertTrue(resultMap.get(RepoService.CREATED).isEmpty());
+        Optional optional = serviceTaskServiceImpl.getServiceTasks().stream().filter(serviceTaskSummary -> serviceTaskSummary.getName().equals(WORKITEMNAME)).findFirst();
+        assertTrue(optional.isPresent());
+    }
+
+    @Test
+    public void testRemoveServiceTask() {
+        List<RepoData> currentServices = new ArrayList<>();
+        loadServices("Test1", currentServices);
+        RepoData rd = currentServices.get(0);
+        ServiceTaskSummary st = new ServiceTaskSummary();
+        st.setName(rd.getName());
+        st.setId(rd.getId());
+        st.setInstalledOn(Collections.emptySet());
+        assertEquals(SERVICETASKSSIZE, serviceTaskServiceImpl.getServiceTasks().size());
+        assertEquals("Test1", serviceTaskServiceImpl.removeServiceTask(st));
+        assertEquals(SERVICETASKSSIZE - 1, serviceTaskServiceImpl.getServiceTasks().size());
```

Comment:
please also add a check that make sure, there is no "Test1" task in the list anymore.

Reply:
Done